packages/backend/src/apps/m365-excel/dynamic-data/list-tables/index.ts

```diff
@@ -0,0 +1,36 @@
+import type { IDynamicData } from '@plumber/types'
+
+import { getM365TenantInfo } from '@/config/app-env-vars/m365'
+
+import { getRegisteredAuthData } from '../../common/auth-data'
+
+const dynamicData: IDynamicData = {
+  name: 'List Tables',
+  key: 'listTables',
+
+  async run($) {
+    const { fileId } = $.step.parameters
+    if (!fileId) {
+      return {
+        data: [],
+      }
+    }
+
+    const authData = getRegisteredAuthData($)
+    const tenant = getM365TenantInfo(authData.tenantKey)
+    const results = await $.http.get<{
+      value: Array<{ id: string; name: string }>
+    }>(
+      `/v1.0/sites/${tenant.sharePointSiteId}/drive/items/${fileId}/workbook/tables?$select=id,name`,
+    )
+
+    return {
+      data: results.data.value.map((entry) => ({
+        name: entry.name,
+        value: entry.id,
+      })),
+    }
+  },
```

Comment:
nit: a more informative error message for dynamic data 
<img width="698" alt="image" src="https://github.com/opengovsg/plumber/assets/10072985/187aeb9f-c206-4fc7-953a-8e0349de6b1f">

Reply:
btw, is list tables not worksheet specific?

Reply:
if not, why do we need this field?